lebedigital/BeamDesign/BeamDesign.py

```diff
@@ -0,0 +1,166 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Fri Sep 16 11:13:33 2022
+
+@author: zv1174
+"""
+import math
+
+def section_dimension_rule_of_thumb(span:float)->tuple:
+    """
+    This is as per eurocode guideline choice of section dimension based on span 
+
+    Parameters
+    ----------
+    span : float
+        beam span.
+
+    Returns
+    -------
+    tuple
+        tuple of breath and depth of the beam.
+
+    """
+    #span/depth ratio for simply supported beam is 15 Therefore,
+    d = 50*round((span/15)/50)
+    #width of beam
+    b = 0.5*d
+    return b,d 
+
+def max_bending_moment_and_shear_force(span:float,load:float,load_type:str) ->tuple:
```

Comment:
I would prefer to directly use width and height as input here. (See previous comment)
In addition, if I understood correctly we are also thinking about including options of I and T beams right?
Would it be possible to parametrize the beam cross section to include those options already (the rectangular one just being a special case of this)?

Reply:
you can directly use that i think there is misunderstanding this is not execution script this are helper functions... 
The design of beam is only valid for singly reinforced rect beam not for I and T beam so those sections wont work. In eurocode 2 if you check for singly reinforced T beam they are designed as rect beam with contribution from flange not computed. Hence they will be always more expensive. For I beam i couldnt find any design guidelines but must then ask someone. But if the section is not designed for compression reinforcement which is the case for singly reinforced section then I beam would also not make any different in terms of amount of steel. As far as I understood we dont want to go towards compression reinforcement or heavily reinforced section for design in final demostrator..

Reply:
It would be helpful if you then could add the execution script, so that we can understand how these will work together.

Reply:
I am not sure in what way the T beam is "allways more expensive". Is it not the point to reduce the amount of used cement? I would assume this could be a good way.
I would assume that for a simple bending load the I beam will be computed the same as a T beam. A difference will be in rotational moments, but I dont think we can about that load case.

Reply:
you are right stress distribution would be different for T beam within section but way eurocode suggest to design is that dont consider flange and thus you reduce T beam to rectangular beam. Hence for design purposes the extra concrete in flange has no role and hence you will get same reinforcement as one for rectangular beam. There can be more detail design with compression reinforcement considered in the section but that is not done in the present script. It is only for special case of singly reinforced concrete beam with rectangular c/s

Reply:
I looked at the test script, this makes it more clear. It would be better to write a function, basically doing what the test is doing and then testing this function. That way we can have a clear overview of the input and output of the final function and directly use it in the workflow.
After a quick search I found there are different cases, according to e.g. (https://rudolfbaumgart.de/scripte/biegebem.pdf). I guess this discussion might be easier in person. For now the rectangle is a good start.

Reply:
> 

Thanks for the ref i see they divided into two cases case 1 they dont consider flange and case 2 where they change to width of the c/s to an equivalent width and then proceed with rectengular section design for now i would suggest to make first series of optimization on rect section and see if something logical is coming out.

Reply:
@raviapatel any progress?